src/biz/metrics_test.go

```diff
@@ -151,3 +154,89 @@ func TestCpuPercent(t *testing.T) {
 	}
 
 }
+
+func TestMetricsFetcher_CPU_LimitCores(t *testing.T) {
+	type args struct {
+		cpu  raw.Metrics
+		json *types.ContainerJSON
+	}
+
+	store, _ := persist.NewFileStore(
+		"test-path",
+		log.NewStdErr(true),
+		1*time.Second)
+
+	tests := []struct {
+		name string
+		args args
+		want CPU
+	}{
+		{
+			name: "LimitCores honors cpu quota even if online CPUs is set",
+			args: args{
+				cpu: raw.Metrics{
+					ContainerID: "test-container",
+					CPU: raw.CPU{
+						OnlineCPUs: 4,
+					},
+				},
+				json: &types.ContainerJSON{
+					ContainerJSONBase: &types.ContainerJSONBase{
+						HostConfig: &container.HostConfig{
+							Resources: container.Resources{
+								NanoCPUs: 500000000,
+							},
+						},
+					},
+				},
+			},
+			want: CPU{
+				LimitCores: 0.5,
+			},
+		},
+		{
+			name: "LimitCores set to OnlineCPUs when no CPU quota",
+			args: args{
+				cpu: raw.Metrics{
+					CPU: raw.CPU{
+						OnlineCPUs: 4,
+					},
+				},
+				json: &types.ContainerJSON{
+					ContainerJSONBase: &types.ContainerJSONBase{
+						HostConfig: &container.HostConfig{},
+					},
+				},
+			},
+			want: CPU{
+				LimitCores: 4,
+			},
+		},
+		{
+			name: "LimitCores set to default runtime.NumCPU() when neither CPU quota or OnlineCPUs set",
+			args: args{
+				cpu: raw.Metrics{
+					CPU: raw.CPU{},
+				},
+				json: &types.ContainerJSON{
+					ContainerJSONBase: &types.ContainerJSONBase{
+						HostConfig: &container.HostConfig{},
+					},
+				},
+			},
+			want: CPU{
+				LimitCores: 2,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mc := &MetricsFetcher{store: store}
+
+			if got := mc.cpu(tt.args.cpu, tt.args.json); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("MetricsFetcher.cpu() = %v, want %v", got, tt.want)
```

Comment:
Nit: as we are only comparing `LimitCores ` here, I would avoid using `reflect.DeepEqual`. What do you think about something like this?

```go
			if got := mc.cpu(tt.args.cpu, tt.args.json); got.LimitCores != want.LimitCores {
			    // ...
```

or using testify:

```go
assert.Equal(t, want.LimitCores, got.LimitCores)
```

Reply:
I forgot to add it in the description... I did it this way as it would be easily extendable in case we want to check all the cpu metrics (which would be goo imo). We would need just to update the test name and add all the rest of metrics. If we want to do it separately you approach is by far easier. WDYT?

Reply:
Well, we would also need to update the storer (to keep other previous values) and both the args and expectation to include the rest of the metrics, so I would rather keep it simple until we extended it. Anyway, it is not important.